### ISSUE TITLE
[JavaScript] LGTM.com - false positive for logical OR assignment

### ISSUE THREAD

**Issue by timvandermeij** [github/codeql#6362]
**Description of the false positive**

Mozilla's PDF.js project recently integrated CodeQL and it doesn't seem to understand logical OR assignment, or the `||=` operator. This operator is explained on https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_OR_assignment. On the link below LGTM states that the `externalListeners` variable is never set, but it clearly is on line https://lgtm.com/projects/g/mozilla/pdf.js/snapshot/02cf4948474989afc65c0b58414fc2dc379119bc/files/web/ui_utils.js?sort=name&dir=ASC&mode=heatmap#L782.

**URL to the alert on the project page on LGTM.com**

https://lgtm.com/projects/g/mozilla/pdf.js/snapshot/02cf4948474989afc65c0b58414fc2dc379119bc/files/web/ui_utils.js?sort=name&dir=ASC&mode=heatmap#x9a849b47d4d84b8e:1

**Comment by timvandermeij** [comment]
For reference, this is not LGTM-specific since we now integrated with GitHub Advanced Security and that shows the same issue (presumably because both use the same CodeQL ruleset); see https://github.com/mozilla/pdf.js/security/code-scanning/110?query=ref%3Arefs%2Fheads%2Fmaster.

**Comment by erik-krogh** [comment]
You're right, that is a false positive. 
I'm looking into it.  

Here is a minimal example that triggers the false positive. 
```JavaScript
function f() {
    let foo;
    if (bar) foo ||= [];
    if (foo);
};
```

**Comment by erik-krogh** [comment]
The FP should be fixed with #6459 

If might take a few weeks for the fix to appear on LGTM / GHAS. 

Again, thanks for the report. 
The fix also fixed a few other FPs in some other projects we hadn't noticed.

**Comment by timvandermeij** [comment]
Glad to help, and thank you for fixing this so quickly!
